### DISCUSSION TITLE
Pinout for 19005 soc module

### DISCUSSION THREAD

**Discussion by DSchndr** [husqvarnagroup/smart-garden-gateway-public#67] [Q&A] [answered]
Hi, could the pinout for that module be published or is it highly confidential? xD
I am kinda "not in the mood" to trace them out with a multimeter in order to add full gpio support to my openwrt build.

**Comment by easybe** [comment, marked as answer]
Hi. Nice to hear that you are running OpenWrt on our hardware. I actually did a [quick hack](https://github.com/husqvarnagroup/openwrt/commits/gardena/eb/mt7688/) some time ago. It would be nice to eventually get the hardware supported upstream.

BTW: I got the overlay file system to work by running the following in U-Boot, other modifications were not needed (but for Linux 6.6 you do need to modify U-Boot).
```
ubi remove overlay; ubi create rootfs_data 0x1e00000 dynamic 4
``` 

Unfortunately, we cannot just share all the documentation. I assume you are working on the DT, so the following snippets should get you going.

![pinout_mt7688_module_int](https://github.com/husqvarnagroup/smart-garden-gateway-public/assets/594536/b8ceff65-b549-49da-b068-238ba66671a5)

![pinout_mt7688_module](https://github.com/husqvarnagroup/smart-garden-gateway-public/assets/594536/6487a660-b9a5-455c-b935-cb75579baddc)

Pin number 1 is in [this image](https://github.com/husqvarnagroup/smart-garden-gateway-public/blob/main/doc/19005-pins.jpg) in the top-right corner.

Let us know if you need further details.

Cheers,
Ezra.

**Comment by DSchndr** [comment]
Haha, did not expect that! :D
Thanks a lot, will finish the devicetree in my free time.
I tried to get it to be "stock compatible" but stumbled across various issues - so after a couple of beers in I tought ["in the bin"](https://www.youtube.com/watch?v=v8CK6CLGTjM) with *that* idea and did it how it is now 
(which should be revertable by just changing the uboot env and flashing stock firmware afterwards - the uart/usb header would also be a clear "warranty void" indicator :) )

Maybe something for the next board revision:
Where poe? it should not add a lot to the bom cost of such an expensive device and would prevent users from plugging in random power supplys (which is the reason why i got some dead gateways for cheap).

Off topic:
It would have been hilarious if I2S_DO/PCMDTX would have been routed to one of the red leds, one idea i have not gotten around to was to just epoxy an optical toslink cable to one of my APs which has it on a led gpio xD
Since I am deeper in the topic (hacking rtk-gps into my mower): does the lemonbeat module just send "tif packets" to the devices?

**Comment by rettichschnidi** [comment]
> does the lemonbeat module just send "tif packets" to the devices?

Regarding tif packets: It just relays what is coming from the cloud.

**Comment by rettichschnidi** [comment]
I guess it is OK to close this. Feel free to reopen if not.

**Comment by DSchndr** [comment]
Ah, yes ofc :)
I actually wanted to close it yesterday but forgot about it.

Thanks for the Info! 
I wondered if i could remote control the new mover over the radio like my older one which has none (for rtk-gps/opencv fencing), nice to know that both use "the same base" :)

**Comment by rettichschnidi** [comment]
@koalatux would know more about that. Maybe even correct me. :D

**Comment by koalatux** [comment]
Depends on the mower. Only Flymo UltraLife (since 2024) and Gardena smart Sileno pro/max/free (since 2025) use Tif to the cloud. But don't take my word for it. (And I do not recommend buying one of those.)
